tracing-mock/src/field.rs

```diff
@@ -80,6 +261,50 @@ impl ExpectedField {
         .and(other)
     }
 
+    /// Indicates that no fields other than those specified should be
+    /// expected.
+    ///
+    /// If additional fields are present on the recorded event or span,
+    /// the expectation will fail.
+    ///
+    /// # Examples
+    ///
+    /// The following test passes despite the recorded event having
+    /// fields that were not expected because `only` was not
+    /// used:
+    ///
+    /// ```
+    /// use tracing_mock::{collector, expect};
+    ///
+    /// let event = expect::event()
+    ///     .with_fields(expect::field("field").with_value(&"value"));
+    ///
+    /// let (collector, handle) = collector::mock().event(event).run_with_handle();
+    ///
+    /// tracing::collect::with_default(collector, || {
+    ///     tracing::info!(field = "value", another_field = 42,);
+    /// });
+    ///
+    /// handle.assert_finished();
+    /// ```
+    ///
+    /// If we include `only` on the `ExpectedField` then the test
+    /// will fail:
```

Comment:
```suggestion
    /// The following example fails because [`ExpectedField::only`] is called:
```

Reply:
I worded this a bit differently because the difference between the 2 examples is now that the second example records an additional field.